load-test/scripts/loadTest.js

```diff
@@ -53,20 +56,53 @@ export default function () {
   check(uploadResult, { "upload status is 200": (r) => isResultSuccess(r)});
 
   if (isResultSuccess(uploadResult)) {
-    let res = null;
-    try {
-      res = ws.connect(WEBSOCKET_URL + authToken, WEBSOCKET_PARAMS, (socket) =>
-        onSocketConnect(socket, requestStartTime, Date.now(), sessionId)
-      );
-    } catch(error) {
-      console.log(`ERROR ${error} for session id ${sessionId}`);
-    }
+    let res = connectToWebsocketWithRetry(authToken, sessionId, requestStartTime);
     check(res, { "websocket status is 101": (r) => r && r.status === 101 });
   } else {
     console.log(`ERROR upload failed for session id ${sessionId}`);
   }
 }
 
+function connectToWebsocketWithRetry(authToken, sessionId, requestStartTime) {
+  let res = null;
+  let tries = 0;
+  let shouldRetry = true;
+  while(tries < 3 && shouldRetry) {
+    if (tries > 0) {
+      // before the first retry sleep for 1 second, before the second retry sleep for 2 seconds.
+      console.log(`RETRY ${tries} for session id ${sessionId}`);
+      sleep(1 * tries);
+    }
+    res = connectToWebsocket(authToken, sessionId, requestStartTime);
+    if (res != null) {
+      shouldRetry = false;
+    } 
+    tries++;
+  }
+  if (tries === 1) {
+    zeroRetries.add(1);
+  } else if (tries === 2) {
+    oneRetry.add(1);
+  } else if (tries === 3) {
+    twoRetries.add(1);
```

Comment:
nit: just a thought, but should we make this more generic by storing the counters in an array? e.g. `retryCounters[tries - 1].add(1)`? may be useful if we ever end up needing more than 3 retries - however if you think that's unlikely, then this is probably easier.

Reply:
good idea, updated!